codegen/service/service_data.go

```diff
@@ -195,6 +203,73 @@ type (
 		// In indicates the request element that holds the credential.
 		In string
 	}
+
+	// ViewedTypeData contains the data to generate types to deal with
+	// view rendering and also the service constructor code to convert
+	// a result type to its viewed counterpart (and vice versa). If ViewType
+	// attribute is set, it denotes a result type having multiple views.
+	// All the fields in the type struct are pointers except for the view
+	// attribute in the viewed result type which is never a pointer.
+	ViewedTypeData struct {
+		// Result type for which the viewed type is generated.
+		*UserTypeData
+		// ViewType contains the data describing a viewed result type.
+		ViewType *UserTypeData
+		// FullRef is the complete reference to the viewed type
+		// (including views package name).
+		FullRef string
+		// Views lists views defined on a result type
+		Views []*ViewData
+		// ToResult is the constructor code to convert a viewed result type
+		// to a result type.
+		ToResult *InitData
+		// ToViewed is the constructor code to convert a result type to a
+		// viewed result type.
+		ToViewed *InitData
+		// Validate is the validations run on the type which is not a
+		// viewed type (i.e., ViewType is not set)
+		Validate string
+	}
+
+	// ViewData contains data about a result type view.
```

Comment:
Would be good to use the `project` nomenclature like is used in the existing code. So there is a result type which can be projected to multiple views. If I understand correctly `ViewData` could then be named `ProjectedData` and `Conversion` should be changed to use `Project`. `ToViewed` might also just be `Project` or `ProjectCode` or `Projection` ... you get the idea :) - Maybe I'm misunderstanding the use case though.